examples/pytorch/image-classification/run_image_classification.py

```diff
@@ -268,9 +274,14 @@ def main():
             "imagefolder",
             data_files=data_files,
             cache_dir=model_args.cache_dir,
-            task="image-classification",
         )
 
+    # Rename image and label columns if needed (e.g. Cifar10)
+    if data_args.image_column_name != "image" and data_args.image_column_name in (dataset["train"].features if "train" in dataset else dataset["validation"].features):
+        dataset = dataset.rename_column(data_args.image_column_name, "image")
+    if data_args.label_column_name != "labels" and data_args.label_column_name in (dataset["train"].features if "train" in dataset else dataset["validation"].features):
+        dataset = dataset.rename_column(data_args.label_column_name, "labels")
```

Comment:
To be consistent with the `audio-classification` task, I think we should do the following:

```suggestion
    dataset_column_names = dataset["train"].column_names if "train" in dataset else dataset["validation"].column_names
    if data_args.image_column_name not in dataset_column_names:
        if "img" in dataset_column_names: 
            logger.info(f"Renaming column img to {data_args.image_column_name}")
            dataset = dataset.rename_column("img", data_args.image_column_name)
        else:
            raise ValueError(
                f"--image_column_name {data_args.image_column_name} not found in dataset '{data_args.dataset_name}'. "
                "Make sure to set `--image_column_name` to the correct image column - one of "
                f"{', '.join(dataset_column_names)}."
        )
    if data_args.label_column_name not in dataset["train"].column_names:
        raise ValueError(
            f"--label_column_name {data_args.label_column_name} not found in dataset '{data_args.dataset_name}'. "
            "Make sure to set `--label_column_name` to the correct text column - one of "
            f"{', '.join(dataset_column_names)}."
        )
```

(Then, we later need to rename `label` to `labels` either via `rename_column` or inside the `train_transforms`/`val_transforms` so that we don't break the script)

Reply:
If the goal is to have something generic, we should precisely avoid using "img" in the example.
`data_args.image_column_name` should not be the name of the new column, we already know it is "image" like in the task template. It should be the name of the column to rename so that users can easily run the script if the image column name is not "img" or "image".